maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/TestSetRunListener.java

```diff
@@ -76,7 +76,7 @@ public TestSetRunListener( ConsoleReporter consoleReporter, FileReporter fileRep
                                StatisticsReporter statisticsReporter, boolean trimStackTrace,
                                boolean isPlainFormat, boolean briefOrPlainFormat )
     {
-        this.consoleReporter = consoleReporter;
+        this.consoleReporter = consoleReporter != null ? consoleReporter : new NullConsoleReporter();
```

Comment:
All of these classes are in `surefire-common` project. Therefor it would be nice not to create a new object 
`new NullConsoleReporter()` but instead keep it as static final constant in abstract Factory in surefire-common.
This would mean that such null-check may be embedded in the factory, e.g. `buildConsoleReporter()`. I would say that the caller should pass non-null consistent object into this class `TestSetRunListener` via constructor and the called class should not make any guess about it (means if == null => guess hide the error). If we accepted such paradigm then all such occurrences in this PR would have to change it.